### PR TITLE
dependency-review: Add `base-ref` config and instructions

### PR DESCRIPTION
**Pull request by jentfoo**
We need to have this configuration on the release branches or merge queue will fail to work correctly.  Currently it is trying to diff to the `master` branch.

### CODE REVIEW

**Review by hugoShaka — APPROVED** [review]

File: .github/workflows/dependency-review.yaml [hunk 1]
```diff
@@ -10,3 +10,5 @@ jobs:
     permissions:
       contents: read
       pull-requests: write
+    with:
+      base-ref: master
```
Comment (reedloden) [inline .github/workflows/dependency-review.yaml:14]:
```suggestion
      base-ref: github.event.pull_request.base.sha || 'master'
```

`master` alone will not work well for PRs that may be based on a different commit. However, `master` is needed for merge queue.

File: .github/workflows/dependency-review.yaml [hunk 2]
```diff
@@ -10,3 +10,5 @@ jobs:
     permissions:
       contents: read
       pull-requests: write
+    with:
+      base-ref: github.event.pull_request.base.sha || 'master'
```
Comment (reedloden) [inline .github/workflows/dependency-review.yaml:14]:
```suggestion
      base-ref: ${{ github.event.pull_request.base.sha }} || 'master'
```

Sorry, typo.

Comment (jentfoo) [reply .github/workflows/dependency-review.yaml:14]: I think it should be `${{ github.event.pull_request.base.sha || 'master' }}`.  I was just about to push but noticed your change.

**Review by ptgott — APPROVED** [review]
👍, assuming Reed's feedback is addressed

File: .github/workflows/dependency-review.yaml [hunk 3]
```diff
@@ -10,3 +10,5 @@ jobs:
     permissions:
       contents: read
       pull-requests: write
+    with:
+      base-ref: ${{ github.event.pull_request.base.sha }} || 'master'
```
Comment (reedloden) [inline .github/workflows/dependency-review.yaml:14]:
```suggestion
      base-ref: ${{ github.event.pull_request.base.sha || 'master' }}
```

I'm just tired today. Sorry.

**Review by reedloden — APPROVED** [review]

**Review by codingllama — APPROVED** [review]

**Review by camscale — APPROVED** [review]
